website/src/_posts/2017-08-28-timely-processing.md

```diff
@@ -248,8 +253,31 @@ new DoFn<Event, EnrichedEvent>() {
 ```
 
 ```py
-# State and timers are not yet supported in Beam's Python SDK.
-# Follow https://issues.apache.org/jira/browse/BEAM-2687 for updates.
+class StatefulBufferingFn(beam.DoFn):
+
+  MAX_BUFFER_SIZE = 500;
+
+  BUFFER_STATE = BagStateSpec('buffer', EventCoder())
+
+  COUNT_STATE = CombiningValueStateSpec('count',
+                                        VarIntCoder(),
+                                        combiners.SumCombineFn())
+
+  def process(self, element,
+              buffer_state=beam.DoFn.StateParam(BUFFER_STATE),
+              count_state=beam.DoFn.StateParam(COUNT_STATE)):
+
+    buffer_state.add(element)
+
+    count_state.add(1)
+    count = count_state.read()
+
+    if count > MAX_BUFFER_SIZE:
+      for event in buffer_state.read():
+        yield event
+
+      count_state.clear()
+      buffer_state.clear()
```

Comment:
Maybe remove the blank line above to make clear that these are part of the `if`?